waspc/data/Generator/templates/react-app/src/auth/logout.js

```diff
@@ -1,9 +1,15 @@
-import { clearLocalStorage } from '../api.js'
 import { invalidateAndRemoveQueries } from '../operations/resources'
+import config from '../config.js'
+import api, { handleApiError } from '../api.js'
 
 export default async function logout() {
-  clearLocalStorage()
-  // TODO(filip): We are currently invalidating and removing  all the queries, but
-  // we should remove only the non-public, user-dependent ones.
-  await invalidateAndRemoveQueries()
+  try {
+    await api.post(config.apiUrl + '/auth/logout')
```

Comment:
I can't remember when we established this try/catch pattern I have to admit, so it might be worth checking that it works ok!

Regarding my initial question, what I was trying to ask is: if there is no connection, how do we logout, since we are now using sessions and not JWTs? With JWTs it is "easy" -> you delete the token from the local storage and voila, you are logged out. But with sessions, I guess you need server? Or can I just delete the cookie locally, what about that?

Reply:
Yeah, we may need to double-check on this try/catch error handling pattern with axios. I can try as time permits.

Correct, with JWTs, since all-state was local, it was pretty easy. I think our happy path should always rely on the server processing the logout. During that, it will set the cookie to empty, and in the past, so it will be expired and removed by browsers. We could have special case logic for logout failing to remove the cookie ourselves, but I think it would be best if we had a general "offline" detection in some way that perhaps called some callback (that may notified the user) and maybe was smarter about certain calls.

So for logout, maybe we do remove the cookie locally, but maybe not (what is the worst case for logout if it cannot complete, so long as the user knows?). For fetching the csrf token, maybe we retry that before we do any other calls. Etc. But that could get complex fast. :/ 

I may just open a new issue for "what if important calls fail" that we can better address holistically there.

Reply:
He he holistically :D.
Ok sounds good! I think the important question you asked here is: what do we expect from logout? With JWTs, answer was that we expect it to delete token from the client's machine. This was considered a security measure, so for example if you are in library, and you login on their public PC somewhere, you can be sure your token got deleted before you left, and you are logged out. So I guess we want something similar -> we want to make sure after we press "logout", that nobody can come after us, visit the web page we were logged in into, and continue from there with our account logged in. However I don't understand the sessions/cookies enough to understand that is needed to ensure that. I was trying to guess that deleting cookie locally is probably enough?

Reply:
Yeah, I agree. I think with how our session is stored inside the cookie, for now, it would be the same as the JWT - just deleting it locally should be fine (so long as the logout function on the server does not have any additional user expectations, like updating the db for audit purposes, etc.). However, when our sessions are stored in the DB in the future, I think that just deleting the cookie is probably not ideal since the logic for sign out may be "sign out _all_ or my active sessions" for some contexts. That would require the server to process and acknowledge.

However, up until the "sign out all sessions" use case, I think our fallback could just be removing the cookie as you noted. 👍🏻

Reply:
Based on our chat, the cross-domain aspect of removing locally will require some thinking. :) We will figure it out!

Reply:
Hooray :D!